src/main/java/org/pih/petl/api/ExecutionContext.java

```diff
@@ -25,6 +32,29 @@ public ExecutionContext(JobExecution jobExecution, PetlJobConfig jobConfig, Appl
         this.applicationConfig = applicationConfig;
     }
 
+    public PetlJobConfig getNestedJobConfig(JsonNode jobTemplate, Map<String, String> replacementVariables) {
+        JsonNode pathNode = jobTemplate.get("path");
+        if (pathNode != null) {
+            String path = pathNode.asText();
+            return applicationConfig.getPetlJobConfig(path);
+        }
+        else {
+            PetlJobConfig nestedJobConfig = new PetlJobConfig();
+            nestedJobConfig.setConfigFile(jobConfig.getConfigFile());
+            nestedJobConfig.setType(jobTemplate.get("type").asText());
+            String configuration = PetlUtil.getJsonAsString(jobTemplate.get("configuration"));
+            Map<String, String> replacements = new HashMap<>(jobConfig.getConfiguration().getVariables());
+            if (replacementVariables != null) {
+                replacements.putAll(replacementVariables);
+            }
+            configuration = StrSubstitutor.replace(configuration, replacements);
+            JobConfiguration jobConfiguration = new JobConfiguration(PetlUtil.readJsonFromString(configuration));
+            jobConfiguration.setVariables(replacements);
+            nestedJobConfig.setConfiguration(jobConfiguration);
+            return nestedJobConfig;
+        }
+    }
```

Comment:
This is a convenience method within the Execution Context to get a nested Job, in order that the job-pipeline and iterating-job implementations can both leverage this to run their nested jobs and take advantage of the same features (like variable replacement).